Refactor : 추가 사용자 서비스 일괄적인 Exception 처리, 변경 application.properties

TrixieL20 opened the pull request:
## 📌 과제 설명
일관된 에러 처리를 위해 코드 수정
application.properties 안에 위치한 중요한 값들 이동

## 👩‍💻 요구 사항과 구현 내용
일관된 에러 처리를 위해 코드 수정
application.properties 안에 위치한 중요한 값들 이동

## ✅ 피드백 반영사항
일관된 에러 처리를 위해 코드 수정 #96
application.properties 안에 위치한 중요한 값들 이동 https://github.com/prgrms-be-devcourse/NBE1_2_Team10/pull/64#discussion_r1787026577

## ✅ PR 포인트 & 궁금한 점

TrixieL20 committed:
Refactor : 추가 exception

일관된 에러 처리를 위한 exception 추가

TrixieL20 committed:
Refactor : 추가 일관된 에러 처리, 수정 application.properties

일관된 에러 처리를 위해 수정
application.properties 안에 위치한 중요한 값들 이동

hgh1472 commented on application/src/main/java/core/application/users/controller/UserController.java:
```diff
@@ -57,8 +56,8 @@ public ApiResponse<Message> login() {
     public ApiResponse<MessageResponseDTO> singUp(@Valid @RequestBody UserRequestDTO userRequestDTO) {
-        MessageResponseDTO msg = userService.signup(userRequestDTO);
-        return ApiResponse.onCreateSuccess(msg);
+        MessageResponseDTO messageResponseDTO = userService.signup(userRequestDTO);
+        return ApiResponse.onCreateSuccess(messageResponseDTO);
```
응답에 `String message`만 들어간다면, api 패키지 내에 Message 객체 사용해주시면 될 것 같아요! 이전에 응답처리할 때 message만 보내주는 경우, `Message`로 통일 시켰습니다!
```suggestion
        return ApiResponse.onCreateSuccess(Message.createMessage("/*Message*/"));
```

hgh1472 reviewed (approved):
수고하셨습니다!

TrixieL20 committed:
Fix : 수정 ApiResponse 파라미터

TrixieL20 committed:
Fix : 수정 잘못된 updateUserInfo 메소드 수정

객체의 존재여부에 따른 에러처리 추가
변경된 회원정보가 아닌 기존 회원 정보가 저장되는 로직 수정

TrixieL20 committed:
Fix : 수정 update 시 비밀번호 암호화 추가

jbw9964 commented on application/src/main/java/core/application/users/service/UserServiceImpl.java:
```diff
@@ -67,54 +71,63 @@ public MessageResponseDTO signup(UserRequestDTO userRequestDTO) {
-                .userName(userRequestDTO.getUserName() != null ? userRequestDTO.getUserName() : originUserEntity.getUserName()) // userName 업데이트
+        UserDTO updatedUserDTO = UserDTO.builder()
+                .userId(originUserEntity.get().getUserId()) // 기존 userId 유지
+                .userPw(userRequestDTO.getUserPw() != null ? userRequestDTO.getUserPw() : originUserEntity.get().getUserPw()) // userPw 업데이트
```
`UserDTO` 클래스 필드에 `@NotNull` 같은 validation 이용하면 편할 것 같습니다.

TrixieL20 replied:
준상님이 코멘트 해주신 부분을 보니 맞네요. update(회원 정보 변경) 시 사용하는 dto와 singup(회원 가입) 시 사용하는 dto를 나눠야겠네요. 회원 정보 변경때는 모두 다 null 이어도 상관없게 dto를 바꿔야겠네요.
dto를 줄이고 싶어서 이렇게 만들었는데 잘못되었네요. 말씀해주셔서 감사합니다!

TrixieL20 replied:
수정 완료했습니다.
205ed36c4f45255f1c687e19dacbbd0a9e874039

jbw9964 reviewed (approved):
LGTM

수고하셨습니다

TrixieL20 committed:
Fix : 추가 회원 정보 수정 DTO

회원 정보 수정 전용 dto가 필요해서 추가함
기존에는 회원 가입과 정보 수정에 동일한 DTO를 사용했지만 notnull 여부와
role 필요 여부에 따라 새 dto가 필요하다고 생각함
role은 관리자만 바꿀 수 있으므로 dto에서 받는 요소에서 삭제함
email은 변경은 불가하지만 token의 이메일과 바꾸고자하는 회원정보의 회원
동일 여부를 판단하기 위해 필요한 요소
